test/unit/ensure_not_used_by_test.rb

```diff
@@ -0,0 +1,40 @@
+require 'test_helper'
+
+class EnsureNotUsedByTest < ActiveSupport::TestCase
+  def setup
+    @org1 = FactoryGirl.create(:organization)
+    @org2 = FactoryGirl.create(:organization)
+    @user = User.create :auth_source => auth_sources(:one), :login => "foo", :mail  => "foo@bar.com"
+    @user.organizations = [@org1]
+  end
+
+  test "#hostgroup_should_not_be_deleted_if_used_by_host_in_user_org" do
+    hostgroup = FactoryGirl.create(:hostgroup, :organizations => [@org1, @org2])
+    host = FactoryGirl.create(:host, :hostgroup => hostgroup, :organization => @org1)
+
+    as_user @user do
+     hostgroup.destroy
+     assert_present Hostgroup.where(:id => hostgroup.id)
```

Comment:
perhaps `assert hostgroup.destroy` instead of checking for presence after (ditto below)